[fastlane] Refactor fastlane runner to make it easy to access from docs

KrauseFx opened the pull request:
Something that will help us with docs

KrauseFx committed:
[fastlane] Refactor fastlane runner to make it easy to access from docs

mfurtak commented on fastlane/lib/fastlane/runner.rb:
```diff
@@ -82,15 +82,28 @@ def available_lanes(filter_platform = nil)
+    def class_reference_from_action_name(method_sym)
+      method_str = method_string_from_method_sym(method_sym)
 
       # First, check if there is a predefined method in the actions folder
```
Comment moved to line 104, so should be deleted here?

mfurtak commented on fastlane/lib/fastlane/runner.rb:
```diff
@@ -82,15 +82,28 @@ def available_lanes(filter_platform = nil)
       method_str = method_sym.to_s
       method_str.delete!('?') # as a `?` could be at the end of the method name
+      return method_str
+    end
```
Could this be simplified to:

```ruby
def method_string_from_method_sym(method_sym)
  # as a `?` could be at the end of the method name
  method_sym.to_s.delete('?')
end
```

mfurtak commented:
Seems good - can you be more specific about why it helps?

KrauseFx commented:
Thanks for the review @mfurtak, I'll update the PR. 

Why do we need this?

We use it when verifying that the code samples in the docs are still working as expected, the method is being called here: https://github.com/fastlane/docs/pull/56/files#diff-b9ec8760f6bfcc35a81f1164b8cf0f5cR40

This is needed to fetch a reference to the action based on the a symbol, to fetch the available parameters

KrauseFx committed:
Worked in feedback from @mfurtak

KrauseFx commented:
Great feedback, thanks @mfurtak, I noticed that I also didn't add any comments to the code base, so I added it, and simplified the code 👍  Let me know if that looks good to you

mfurtak commented:
Great comments! LGTM 👍
